### PR TITLE
Make language server dependencies optional via parameter builds

### PR DESCRIPTION
**Pull request by Cypher1**

### CODE REVIEW

**Comment by shans** [comment]
This looks like a good approach. You were worried about brittleness - is that because the deps need to be passed in explicitly?

**Comment by Cypher1** [comment]
I was previously worried that the wrong versions of the deps were going to be installed, it turns out npm supports optional dependencies and package-lock can keep track of the appropriate versions to install without needing them to be installed.

File: src/tools/aml-language-server/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+const path = require('path');
+const webpack = require('webpack');
+
+const buildDir = './build/tools/aml-language-server';
+
+// Package Tools CLIs into self contained .js files to be used with plain node.
```
Comment (piotrswigon) [inline src/tools/aml-language-server/webpack.config.js]: Update?

File: tools/sigh.ts [hunk 2]
```diff
@@ -63,8 +63,12 @@ import * as AstNode from '../../runtime/manifest-ast-nodes.js';
   }]
 };
 
+const build = buildPath('.');
+const webpack = webpackPkg('webpack');
+const webpackTools = webpackPkg('webpack-tools');
+
 const steps: {[index: string]: ((args?: string[]) => boolean)[]} = {
-  languageServer: [peg, build, webpackTools, languageServer],
+  languageServer: [peg, build, buildPath('./src/tools/aml-language-server', ['vscode-jsonrpc', 'vscode-languageserver']), webpackPkg('webpack-languageserver'), languageServer],
```
Comment (piotrswigon) [inline tools/sigh.ts]: I would extract `buildPath('./src/tools/aml-language-server', ['vscode-jsonrpc', 'vscode-languageserver'])` to a const as you did for `buildPath('.')`, will be easier to read IMO

File: tools/sigh.ts [hunk 3]
```diff
@@ -326,22 +330,27 @@ function railroad(): boolean {
...
-  if (!tsc()) {
-    console.log('build::tsc failed');
-    return false;
-  }
+function buildPath(path: string, deps?: string[]): () => boolean {
+  return () => {
+    if (!tsc(path)) {
+      console.log('build::tsc failed');
+      if (deps && deps.length > 0) {
+        console.log(`The following dependencies may be required${deps.map(s=>` ${s}`)}`);
```
Comment (piotrswigon) [inline tools/sigh.ts]: Nice

File: src/tools/aml-language-server/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (piotrswigon) [inline src/tools/aml-language-server/tsconfig.json]: What's the motivation for having a different ts config? Do you need more relaxed one?

**Pull request merged**